### PR TITLE
Record your work search survey

### PR DESCRIPTION
**Pull request by lukasz-kaniowski**
@jessstubenbord created a survey view

I've added test for it later. I've tried to redesign tests a bit so there will be less work next time we have new survey.

### CODE REVIEW

**Comment by jessstubenbord** [comment]
@lukasz-kaniowski this content has been signed off and is ready to merge once the tests are set

File: app/controllers/survey-controller.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ router.get('/:type/:ref', (req, res, next) => {
         .then(survey => {
           const options = new OptionsWithLabels(survey.get('definition')).options;
 
-          res.render('survey', {
+          res.render(req.params.type, {
```
Comment (nathancashmoredwp) [inline app/controllers/survey-controller.js]: Should this be in a different PR

File: app/views/ewycd.mustache [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
     <div class="form-group">
 
-      <h3 class="form-label-bold text" data-test-id="question-why-types">{{labels.whyTypes}}</h3>
+      <h3 class="form-label-bold text" data-test-id="question-whyTypes">{{labels.whyTypes}}</h3>
```
Comment (nathancashmoredwp) [inline app/views/ewycd.mustache]: WYCD verses RYWS...? another PR ?

File: db/migrations/20160908142125_ewycd-2-definition.js [hunk 3]
```diff
@@ -1,56 +1,61 @@
 /* eslint-disable quote-props, max-len */
```
Comment (nathancashmoredwp) [inline db/migrations/20160908142125_ewycd-2-definition.js]: As above.. should be in Amendments to EWYCD feedback

File: test/common/page_objects/ewycd-survey-page.js [hunk 4]
```diff
@@ -11,7 +11,6 @@ class EWYCDSurveyPage extends BaseSurveyPage {
     this.SEARCH_TERMS_HELPED_PANEL = '#whyTypesSearchTermsHelped-panel';
     this.OTHER_HELPED_PANEL = '#whyTypesOtherHelped-panel';
     this.OTHER_REASON_PANEL = '[data-test-id="other-reason-panel"]';
-    this.QUESTION_WHY_TYPES = '[data-test-id="question-why-types"]';
```
Comment (nathancashmoredwp) [inline test/common/page_objects/ewycd-survey-page.js]: As above.... EWYCD ?

File: test/common/page_objects/survey-page.js [hunk 5]
```diff
@@ -11,6 +11,11 @@ class SurveyPage {
   getQuestionValues() {
     return this.browser.queryAll('[data-test-id^="question-"]').map(el => el.textContent.trim());
   }
+
```
Comment (nathancashmoredwp) [inline test/common/page_objects/survey-page.js]: As above .. lots of different changes wrapped all together in here

**Comment by nathancashmoredwp** [comment]
There is loads of stuff all crammed in here... think it would be better to break them out so we have better visibility of what is changes.  Especially as one survey is now live.  What do you think ?

File: app/controllers/survey-controller.js (hunk 1 above)
Comment (lukasz-kaniowski) [inline app/controllers/survey-controller.js]: Controller need to be able to handle `ryws` type. Either duplicate controller or add parametrization. This need to be here.

File: db/migrations/20160908142125_ewycd-2-definition.js (hunk 3 above)
Comment (lukasz-kaniowski) [inline db/migrations/20160908142125_ewycd-2-definition.js]: there is no amendments here. just refactoring internal structure to support tests.

**Comment by lukasz-kaniowski** [comment]
I can I guess. Lot of those changes are in here so I don't write test only for sake of refactoring them 15minutes later. 
I will move the one that are not required by the new test but could be done on their own. 

Did you had a chat with Gareth by any chance - you have that "tab over comma" flavor ;)

**Comment by nathancashmoredwp** [comment]
All looks good, just the tests based on what is already in PR #40 ... would these tests override what is in that or should they also disappear from here ?
👍

**Comment by lukasz-kaniowski** [comment]
Tests will disappear from here. Prs #37 #38 #39 #40 are delivering small incremental changes that initially ware part of this pr.

**Comment by nathancashmoredwp** [comment]
👍  Neat.
